django/contrib/postgres/fields/citext.py

```diff
@@ -12,12 +25,54 @@ def db_type(self, connection):
 
 
 class CICharField(CIText, CharField):
-    pass
+    system_check_deprecated_details = {
+        "msg": (
+            "django.contrib.postgres.fields.CICharField is deprecated. Support for it "
+            "(except in historical migrations) will be removed in Django 5.1."
+        ),
+        "hint": (
+            'Use CharField(db_collation="…") with a case-insensitive non-deterministic '
+            "collation instead."
+        ),
+        "id": "fields.W905",
+    }
+
+    def __init__(self, *args, **kwargs):
+        with ignore_warnings(category=RemovedInDjango51Warning):
+            super().__init__(*args, **kwargs)
```

Comment:
Added to avoid `CIText` deprecation warnings when instantiating `CICharField` (the same in `CIEmailField` and `CITextField`).